File: 프로그래머스/lv2/17680. ［1차］ 캐시/［1차］ 캐시.py
from collections import deque
def solution(cacheSize, cities):
    answer = 0
    cache = deque()
    if cacheSize == 0:
        return 5*len(cities)
    for city in cities:
        city = city.lower()
        #만약 캐시가 있으면
        if city in cache:
            answer += 1
            cache.remove(city)
            cache.append(city)
        else:
            if len(cache) == cacheSize:
                cache.popleft()
            cache.append(city)
            answer+=5

    return answer

File: 프로그래머스/lv2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;
class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        int time = 0;
        Deque<String> cache = new ArrayDeque<String>();
        if(cacheSize == 0){
            time = 5 * cities.length;
            return time;
        }
        for(int i=0; i<cities.length; i++){
            String city = cities[i].toLowerCase();
            if(cache.contains(city)){
                time += 1;
                cache.remove(city);
                cache.add(city);
            }
            else{
                if(cache.size() == cacheSize){
                    cache.removeFirst();
                }
                cache.add(city);
                time += 5;
            }
        }
        return time;
    }
}